### PR TITLE
Port `is_valid_attribute` to rust

### PR DESCRIPTION
**Pull request by pwoolcoc**
See #334

### CODE REVIEW

**Comment by pwoolcoc** [comment]
low-hanging fruit, but it didn't have any cpp tests so I added some rust tests

File: rust/libnewsboat/src/utils.rs [hunk 1]
```diff
@@ -200,5 +218,34 @@ mod tests {
             assert!(is_valid_color(color));
         }
     }
+
+    #[test]
+    fn t_is_valid_attribte() {
```
Comment (Minoru) [inline rust/libnewsboat/src/utils.rs]: Please port this test to C++, so we can be sure that FFI didn't break anything.

For functions without tests, it's best to write tests beforehand to make sure that they actually pass on the original C++ implementation. In this case, the function is pretty trivial, so don't bother with that; but please keep it in mind for the future.

**Comment by rnestler** [comment]
Oh I started implementing this as well because I wasn't aware of this PR: https://github.com/rnestler/newsboat/commit/44e30d7d07530634b3a45345511a704268d56783
One questions: Why do you use a `HashSet`? An array is much simpler and not really slower which such a low amount of elements.

**Comment by Minoru** [comment]
:wave: Hey @pwoolcoc, looks like this PR stalled. Do you mind if I apply (my own) review suggestions and then merge it?

**Comment by pwoolcoc** [comment]
@Minoru hey, sorry, yea sure.

**Comment by Minoru** [comment]
Out of curiosity, I benchmarked both HashSet and array implementations with Criterion, and got this graph:
gnuplot svg

Array's runtimes are so small and tight that they barely show in the top left (they're around 12ns, with variance of half a nanosecond).

`is_valid_attribute` isn't in a critical path of any kind, but since @rnestler already wrote a fast implementation and I already benchmarked it, I don't see a reason not to merge it. I still used your tests, @pwoolcoc, so your efforts weren't wasted either. Thank you both!

Closed via https://github.com/newsboat/newsboat/commit/7d201cf3c7db08f9ddb195e1c88c46fca0f1a5a2. (I rewrote `is_valid_color`, too)

**Pull request closed without merging**